layers/+lang/haskell/packages.el

```diff
@@ -67,123 +74,145 @@
+                     ('ormolu '(haskell-mode))
+                     ('stylish-haskell '(haskell-mode))
+                     (_))))
+       (spacemacs/add-to-hooks 'format-all-mode modes
+        (dolist (mode modes)
+          (spacemacs/set-leader-keys-for-major-mode mode
+            "==" 'format-all-buffer
+            "T=" 'format-all-mode))))
+      ;; also set lsp formattign provider
+      ;; NOTE: except of 'hindent which is not currently supported (April 2021)
+      (when (and (eq haskell-completion-backend 'lsp)
+                 (memq haskell-formatter-backend
+                       '(brittany ormolu stylish-haskell fourmolu floskell)))
+        (setq lsp-haskell-formatting-provider (symbol-name haskell-formatter-backend))))))
+
+
+;; haskell-mode
 
 (defun haskell/init-haskell-mode ()
   (use-package haskell-mode
     :defer t
-    :init
-    (progn
-      (add-hook 'haskell-mode-local-vars-hook #'spacemacs-haskell//setup-backend)
-      (add-hook 'haskell-literate-mode-local-vars-hook #'spacemacs-haskell//setup-backend)
-      ;; renamed as of 04/2020, delete in due course
-      (add-hook 'literate-haskell-mode-local-vars-hook #'spacemacs-haskell//setup-backend)
-
-      (defun spacemacs//force-haskell-mode-loading ()
-        "Force `haskell-mode' loading when visiting cabal file."
-        (require 'haskell-mode))
-      (add-hook 'haskell-cabal-mode-hook
-                'spacemacs//force-haskell-mode-loading)
-
-      ;; Haskell cabal files interact badly with electric-indent-mode
-      ;; note: we cannot add this hook in :config, since haskell-mode might
-      ;; only be loaded after cabal-mode hooks are already run (see add-hook above)
-      (add-hook 'haskell-cabal-mode-hook #'spacemacs-haskell//disable-electric-indent)
-
-      (setq
-       ;; Use notify.el (if you have it installed) at the end of running
-       ;; Cabal commands or generally things worth notifying.
-       haskell-notify-p t
-       ;; Remove annoying error popups
-       haskell-interactive-popup-errors nil
-       ;; Better import handling
-       haskell-process-suggest-remove-import-lines t
-       haskell-process-auto-import-loaded-modules t
-       ;; Disable haskell-stylish-on-save, as it breaks flycheck highlighting.
-       ;; NOTE: May not be true anymore - taksuyu 2015-10-06
-       haskell-stylish-on-save nil))
+    :hook
+    (((haskell-mode-local-vars haskell-literate-mode-local-vars) . spacemacs-haskell//setup-backend)
+     (haskell-cabal-mode . spacemacs//force-haskell-mode-loading)
+     ((haskell-mode haskell-cabal-mode) . spacemacs//haskell-disable-electric-indent))
+    :custom
+    (haskell-notify-p t "Notify with notify.el, if installed, after running Cabal commands and etc")
+    (haskell-interactive-popup-errors nil "Remove annoying error popups")
+    (haskell-process-suggest-remove-import-lines t "Show unused imports")
+    (haskell-process-auto-import-loaded-modules t "Auto-import module")
```

Comment:
`lsp` users probably don't want any of the `haskell-process-` variables set, since they won't be using that!